src/github.com/getlantern/peerscanner/web.go

```diff
@@ -51,36 +56,46 @@ func startHttp() {
 // register is the entry point for peers registering themselves with the service.
 // If peers are successfully vetted, they'll be added to the DNS round robin.
 func register(resp http.ResponseWriter, req *http.Request) {
-	name, ip, port, err := getHostInfo(req)
-	if err == nil && port != 443 {
-		err = fmt.Errorf("Port %d not supported, only port 443 is supported", port)
+	name, ip, port, supportedFronts, err := getHostInfo(req)
+	if err == nil && !(port == "80" || port == "443") {
+		err = fmt.Errorf("Port %d not supported, only ports 80 and 443 are supported", port)
```

Comment:
In order to sit behind Cloudfront, flashlight servers currently need to listen for unencrypted connections (e.g. in port 80).

In order to serve to get mode clients directly via IP (i.e., no domain fronting), flashlight servers would need to listen through HTTPS.

Currently, flashlight servers can only listen through either one or the other.  So if we got blocked right now, we'd need different pools of servers for both purposes.  This is pretty much what we have now with flashlight servers vs old-style fallbacks.  It would be more efficient to serve in both ways from the same pool of servers.

The question is: in the long run, do we prefer to keep the hop between the CDN and flashlight servers encrypted?  If so, I think we need to get a wildcard CA-signed cert for (e.g.) flashlightproxy.org, so Cloudfront will serve HTTPS to them.

If we are happy with having that hop unencrypted (saves some CPU; may have security repercussions, but out of our threat model), then we should change flashlight servers so they can listen on both ports.

I think both changes are rather easy to implement, so it's just a matter of deciding whether we want the CDN<->flashlight server hop encrypted or not.